app/models/content_item.rb

```diff
@@ -152,9 +163,17 @@ def includes_auth_bypass_id?(auth_bypass_id)
   end
 
   def register_routes(previous_item: nil)
-    return if self.schema_name.start_with?("placeholder")
-    return if previous_item && previous_item.route_set == self.route_set
-    self.route_set.register!
+    return unless should_register_routes?(previous_item: previous_item)
+
+    3.times do
+      begin
+        route_set.register!
+        break
+      rescue GdsApi::BaseError => e
+        last_exception = e
+      end
+      raise last_exception
```

Comment:
If `route_set.register!` raises an exception the first time, is the intention to retry 2 more times? In which case, do we want a test in the test suite to ensure that we are retrying?

The underlying question as a ruby neophyte - does `raise` break out of `3.times` prematurely?